www/roster/main.rb

```diff
@@ -167,6 +167,14 @@
   _html :podlings
 end
 
+# gives a list of podlings that would report in a given month based on current status
+get '/podlings/reporting/:month.json' do |month|
```

Comment:
A sinatra application generally require (in addition to your script): a config.ru file, a Gemfile, and a tmp dir.  Such applications are also long running: the process is kept up between requests and only torn down after significant inactivity.  The podling management tool you envision probably should be such a tool.

CGI's scripts that run as a separate process per request.  They also are easier to get started with.  Drop the following file in `whimsy/www/incubator/reporting.cgi` and chmod it to 755:

```
#!/usr/bin/env ruby

$LOAD_PATH.unshift File.realpath(File.expand_path('../../../lib', __FILE__))
require 'wunderbar'
require 'whimsy/asf'

month =  ENV['PATH_INFO'].sub('/', '')

_json do
  ASF::Podling.list
    .select { |p| p.status == 'current' }
    .select { |p| (p.reporting.include? month if p.reporting) ||
                  (p.monthly.include? month if p.monthly) }
    .map { |p| { id: p.id, name: p.display_name, list: p.dev_mail_list } }
end
```

Try it out with http://whimsy.local/incubator/reporting/January

Getting started by writing a big incubator management application may not be the best approach unless you are the type of person that prefers to jump in the deep end.  :-)  Converting the script that you have that sends out reminders to a whimsy/wunderbar application might be an easier first step.

If you point me to the source, I can help; but in the end what you chose to work on and how you chose to implement it is up to you.